l5kit/l5kit/dataset/agent.py

```diff
@@ -11,6 +11,9 @@
 from .ego import EgoDataset
 from .select_agents import TH_DISTANCE_AV, TH_EXTENT_RATIO, TH_YAW_DEGREE, select_agents
 
+# WARNING: changing this value impact the number of instances selected for both train and inference!
+MIN_FRAME_HISTORY = 10  # minimum number of frames an agents must have in the past to be picked
```

Comment:
Is it possible to have the MIN_FRAME_FUTURE as well? Sorry for flip-floping here, but it might be good if we can reproduce the current behaviour in the future if needed. 

Could you extract the other "magic constants" like the distance threshold and the box extent and yaw changes?

It would be also worth considering putting them into config.yml. Then we can change them without having to bake new versions of master. We can also then have a separate setting for training config and for evaluation config (the latter should be kept constant and submitted to the repo).

Reply:
I can add the `MIN_FRAME_FUTURE`. Should I set it to default 0 then?
For the magic constants I'm nor really sure though. They are defined in the script that use them, where would you put them? 

I'm not sure about that...it depends on what we think these values are. If they are something that should never be touched, then config is not really the right place imo.

Reply:
Fair, let's keep it this way. I can see value for changing this for training, but now that I think about it again, we can use the availability masks in GT to do this filtering, right?

Reply:
the mask can always be passed by the caller, so I would say yes